src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -83,54 +64,69 @@ public ViewerConfiguration LoadConfig()
         [Route("loadFileTree")]
         public HttpResponseMessage loadFileTree(PostedDataEntity postedData)
         {
-            string relDirPath = "";
-            // get posted data
-            if (postedData != null)
+            try
+            {
+                List<FileDescriptionEntity> filesList = new List<FileDescriptionEntity>();
+                if (!string.IsNullOrEmpty(globalConfiguration.Viewer.GetFilesDirectory()))
+                {
+                    filesList = this.LoadFiles();
+                }
+                return Request.CreateResponse(HttpStatusCode.OK, filesList);
+            }
+            catch (Exception ex)
             {
-                relDirPath = postedData.path;
+                return Request.CreateResponse(HttpStatusCode.OK, new Resources().GenerateException(ex));
             }
-            // get file list from storage path
-            FileListOptions fileListOptions = new FileListOptions(relDirPath);
-            // get temp directory name
-            string tempDirectoryName = new ViewerConfig().CacheFolderName;
-            try
+        }
+
+        /// <summary>
+        /// Load documents
+        /// </summary>
+        /// <returns>List[FileDescriptionEntity]</returns>
+        public List<FileDescriptionEntity> LoadFiles()
+        {
+            var currentPath = globalConfiguration.Viewer.GetFilesDirectory();
+            List<string> allFiles = new List<string>(Directory.GetFiles(currentPath));
+            allFiles.AddRange(Directory.GetDirectories(currentPath));
+            List<FileDescriptionEntity> fileList = new List<FileDescriptionEntity>();
+
+            // TODO: get temp directory name
+            string tempDirectoryName = "temp";
+
+            allFiles.Sort(new FileNameComparator());
+            allFiles.Sort(new FileDateComparator());
+
+            foreach (string file in allFiles)
             {
-                List<FileDescriptionEntity> fileList = new List<FileDescriptionEntity>();
-                if (!String.IsNullOrEmpty(globalConfiguration.Viewer.GetFilesDirectory()))
+                FileInfo fileInfo = new FileInfo(file);
+                // check if current file/folder is hidden
+                if (tempDirectoryName.Equals(Path.GetFileName(file)) ||
+                    fileInfo.Attributes.HasFlag(FileAttributes.Hidden) ||
+                    Path.GetFileName(file).Equals(Path.GetFileName(globalConfiguration.Viewer.GetFilesDirectory())))
                 {
-                    FileListContainer fileListContainer = this.GetHandler().GetFileList(fileListOptions);
-                    // parse files/folders list
-                    foreach (FileDescription fd in fileListContainer.Files)
+                    // ignore current file and skip to next one
+                    continue;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this redundant jump.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5242614)